### PR TITLE
Add tests for prngPrecompileContract and exchangeRatePrecompileContract

### LINKED ISSUE #6782: HIP-584: Add acceptance tests for PrngSystemPrecompiledContract
### Problem

Missing acceptance test for PrngSystemPrecompiledContract

### Solution

Add new tests for PrngSystemPrecompiledContract.

### Alternatives

_No response_

### LINKED ISSUE #6784: HIP-584: Add acceptance tests for ExchangeRatePrecompiledContract
### Problem

Currently there are not acceptance tests that cover ExchangeRatePrecompiledContract.

### Solution

Add new acceptance tests for ExchangeRatePrecompiledContract.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by yiliev0**
**Description**: 

This PR introduces acceptance tests for the eth_estimateGas functionality, specifically targeting the pseudoRandomSeed (PRNG) and exchangeRate precompiled contracts.

**Note:**
For the PRNG tests to function correctly, PR #6668 must be merged.

**Related issue(s)**:

Fixes #6782
Fixes #6784

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/EstimatePrecompileFeature.java [hunk 1]
```diff
@@ -1912,6 +1958,8 @@ private enum ContractMethods {
         DISSOCIATE_AND_ASSOCIATE("dissociateAndAssociateTokenExternal", 1434814),
         DISSOCIATE_TOKEN("dissociateTokenExternal", 729428),
         DISSOCIATE_TOKENS("dissociateTokensExternal", 730641),
+        EXCHANGE_RATE_TINYCENTS_TO_TINYBARS("tinycentsToTinybars", 24833),
+        EXCHANGE_RATE_TINYBARS_TO_TINYCENTS("tinybarsToTinycents", 24833),
```
Comment (attest08) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/EstimatePrecompileFeature.java:1962]: Is this: 24833 the correct value? It is exactly the same for  EXCHANGE_RATE_TINYCENTS_TO_TINYBARS and  EXCHANGE_RATE_TINYBARS_TO_TINYCENTS

File: hedera-mirror-test/src/test/resources/solidity/contracts/EstimatePrecompileContract.sol [hunk 2]
```diff
@@ -379,4 +383,39 @@ contract EstimatePrecompileContract is HederaTokenService, ExpiryHelper, KeyHelp
         handleResponseCode(responseCode);
         return approved;
     }
+
+    function getPseudorandomSeed() external returns (bytes32 randomBytes) {
```
Comment (attest08) [inline hedera-mirror-test/src/test/resources/solidity/contracts/EstimatePrecompileContract.sol:387]: getPseudoRandomSeed - camelCase

**Comment by attest08** [comment]
LGTM

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/EstimatePrecompileFeature.java (hunk 1 above)
Comment (yiliev0) [reply hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/EstimatePrecompileFeature.java]: Tinybars to tinycents was slightly lower. Changed.

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/EstimatePrecompileFeature.java [hunk 3]
```diff
@@ -1938,6 +1986,8 @@ private enum ContractMethods {
         NESTED_FREEZE_UNFREEZE("nestedFreezeUnfreezeTokenExternal", 54548),
         NESTED_GRANT_REVOKE_KYC("nestedGrantAndRevokeTokenKYCExternal", 54516),
         OWNER_OF("getOwnerOf", 27271),
+        PSEUDO_RANDOM_SEED("getPseudorandomSeed", 36729),
+        PSEUDO_RANDOM_NUMBER("getPseudorandomNumber", 36729),
```
[the remark this answers is not in the document]
Comment (yiliev0) [reply hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/EstimatePrecompileFeature.java]: They are the same as they expose the same precompile.

File: hedera-mirror-test/src/test/resources/solidity/contracts/EstimatePrecompileContract.sol (hunk 2 above)
Comment (yiliev0) [reply hedera-mirror-test/src/test/resources/solidity/contracts/EstimatePrecompileContract.sol]: Done.

Comment (yiliev0) [reply hedera-mirror-test/src/test/resources/solidity/contracts/EstimatePrecompileContract.sol]: Nevermind, we should follow the naming convention based in the HIP -> https://hips.hedera.com/hip/hip-351.

**Review by steven-sheehy — APPROVED** [review]
LGTM

File: hedera-mirror-test/src/test/resources/solidity/contracts/EstimatePrecompileContract.sol [hunk 4]
```diff
@@ -18,6 +18,10 @@ contract EstimatePrecompileContract is HederaTokenService, ExpiryHelper, KeyHelp
     int32 decimals = 8;
     bool freezeDefaultStatus = false;
 
+    address constant PRECOMPILE_ADDRESS = address(0x169);
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-test/src/test/resources/solidity/contracts/EstimatePrecompileContract.sol:21]: We can enhance the naming of these fields by noting what type of precompile sits behind these values, e.g. `PRNG_PRECOMPILE_ADDRESS`, `EXCHANGE_RATE_PRECOMPILE_ADDRESS`

Comment (yiliev0) [reply hedera-mirror-test/src/test/resources/solidity/contracts/EstimatePrecompileContract.sol]: Done.

**Review by steven-sheehy — APPROVED** [review]

**Pull request merged**